src/java.base/share/classes/java/lang/Math.java

```diff
@@ -1007,6 +1002,58 @@ public static long multiplyExact(long x, long y) {
         return r;
     }
 
+    /**
+     * Returns the quotient of the arguments, throwing an exception if the
+     * result overflows an {@code int}.  Such overflow occurs in this method if
+     * {@code x} is {@link Integer#MIN_VALUE} and {@code y} is {@code -1}.
+     * In contrast, if {@code Integer.MIN_VALUE / -1} were evaluated directly,
+     * the result would be {@code Integer.MIN_VALUE} and no exception would be
+     * thrown.
+     * <p>
+     * If {@code y} is zero, overflow also occurs as in direct evaluation and
```

Comment:
To clarify my earlier comments, I don't think the divide-by-zero case should be described as "overflow". An overflow condition generally could be handled if the result format had greater range. That is not true of dividing by an exact zero.

However, the methods should still throw an exception for divide by zero.

Reply:
@bplb More specifically, a int/long division by 0 throws ArithmeticException("/ by zero")